### PR TITLE
[aot] Emit ref instances of the InternalArray methods only in corlib.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/mini/aot-compiler.c [hunk 1]
```diff
@@ -4861,7 +4861,8 @@ add_generic_class_with_depth (MonoAotCompile *acfg, MonoClass *klass, int depth,
 			if (strstr (method->name, name_prefix)) {
```
Comment (jaykrell) [inline mono/mini/aot-compiler.c]: Should this be strncmp? strstr is much slower, and the term `prefix` here suggests strstr is overkill.

**Comment by jaykrell** [comment]
Regarding strncmp, will presence of underscore at start be the same on each?

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by kumpera** [comment]
Linux AArch64 fails finalizer-thread.exe with: ```
* Assertion at unwind.c:640, condition `cfa_reg != -1' not met

* Assertion at mini-exceptions.c:1202, condition `unwind_options == MONO_UNWIND_NONE' not met
```

i386 OSX fails System.Runtime.Caching with this:
```
Results saved as /Users/builder/jenkins/workspace/test-mono-pull-request-i386-osx/mcs/class/System.Runtime.Caching/TestResult-net_4_x.xml.
mono(53061,0xb0956000) malloc: *** error for object 0x4e2f00: pointer being freed was not allocated
*** set a breakpoint in malloc_error_break to debug
Memory around native instruction pointer (0xa1750b12):
0xa1750b02  ff ff 83 ec 04 56 57 53 e8 3d df 02 00 83 c4 10  .....VWS.=......
0xa1750b12  0f 0b 55 89 e5 53 57 56 83 ec 0c e8 00 00 00 00  ..U..SWV........
0xa1750b22  5b 8b 75 08 e8 a7 da 02 00 89 c7 85 ff 74 22 8d  [.u..........t".
0xa1750b32  45 0c 89 45 f0 83 ec 04 50 56 57 e8 ae da 02 00  E..E....PVW.....

Native stacktrace:

	0   mono                                0x00130baf mono_dump_native_crash_info + 767
	1   mono                                0x000dd4b3 mono_handle_native_crash + 179
	2   mono                                0x000304df mono_sigill_signal_handler + 47
	3   libsystem_platform.dylib            0xa18cae5b _sigtramp + 43
	4   ???                                 0xffffffff 0x0 + 4294967295
	5   libsystem_c.dylib                   0xa1750a4c __abort + 0
	6   libsystem_malloc.dylib              0xa18478e7 free + 529
	7   CoreFoundation                      0x942c84da __exceptionFinalize + 74
	8   CoreFoundation                      0x942a5d72 __CFTSDFinalize + 114
	9   libsystem_pthread.dylib             0xa18d5ac5 _pthread_tsd_cleanup + 596
	10  libsystem_pthread.dylib             0xa18d57e2 _pthread_exit + 137
	11  libsystem_pthread.dylib             0xa18d6081 pthread_exit + 33
	12  mono                                0x00305151 mono_threads_platform_exit + 17
	13  mono                                0x0030371c mono_thread_info_exit + 60
	14  mono                                0x002451f0 start_wrapper + 816
	15  libsystem_pthread.dylib             0xa18d4047 _pthread_body + 184
	16  libsystem_pthread.dylib             0xa18d3f8f _pthread_body + 0
	17  libsystem_pthread.dylib             0xa18d384a thread_start + 34
```
Windows x64 fails System with: ```
***** MonoTests.Microsoft.CSharp.CSharpCodeProviderTest.CompileFromDomBatch_NotInMemory
* Assertion: should not be reached at d:\j\workspace\x\mono\sgen\sgen-scan-object.h:91

make[2]: *** [../../build/tests.make:219: run-test-lib] Error 1
```

None of those are related to a change to full-aot

**Pull request merged**